src/test/java/com/google/sps/DataServletTest.java

```diff
@@ -0,0 +1,229 @@
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import javax.servlet.http.HttpServletRequest;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+public final class DataServletTest {
+
+  private static final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  private addTermRating newTermRating;
+
+  @BeforeEach
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+    helper.setUp();
+    newTermRating = new addTermRating();
+  }
+
+  @AfterEach
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Mock HttpServletRequest request;
+
+  @Mock HttpServletRequest requestB;
+
+  @Test
+  public void AddingNewTermRatingNoSentimentAnalysisNoTranslation() {
+
+    DatastoreService db = DatastoreServiceFactory.getDatastoreService();
+    request = Mockito.mock(HttpServletRequest.class);
+
+    // First Term Rating for this term.
+    when(request.getParameter("class-input")).thenReturn("I really like this class.");
+    when(request.getParameter("rating-class")).thenReturn("4");
+    when(request.getParameter("hoursOfWork")).thenReturn("8");
+    when(request.getParameter("difficulty")).thenReturn("4");
+    when(request.getParameter("prof-input")).thenReturn("The professor was okay.");
+    when(request.getParameter("rating-professor")).thenReturn("3");
+    when(request.getParameter("languages")).thenReturn("false");
+    when(request.getParameter("ID")).thenReturn("numberOneId");
+    Entity termEntity = new Entity("Term");
+    Key termKey = termEntity.getKey();
+    when(request.getParameter("CourseClass"))
+        .thenReturn(Course("Spring 2020", "Smith", 12, termKey));
+
+    newTermRating.addTermRating(request);
+
+    Entity termRatingQuery =
+        queryEntities("Term Rating", reviewer - id, request.getParameter("reviewer-id")).get(0);
+    Entity expectedTermRatingEntity = new Entity("Term Rating");
+    expectedTermRatingEntity.setProperty("comments-class", "I really like this.");
+    assertEquals(
+        expectedTermRatingEntity.getProperty("comments-class"),
+        termRatingQuery.getProperty("comments-class"));
+    expectedTermRatingEntity.setProperty("reviewer-id", "numberOneId");
+    assertEquals(
+        expectedTermRatingEntity.getProperty("reviewer-id"),
+        termRatingQuery.getProperty("reviewer-id"));
+    expectedTermRatingEntity.setProperty("perception-class", 4);
+    assertEquals(
+        expectedTermRatingEntity.getProperty("perception-class"),
+        termRatingQuery.getProperty("perception-class"));
+    expectedTermRatingEntity.setProperty("hours", 8);
+    assertEquals(
+        expectedTermRatingEntity.getProperty("hours"), termRatingQuery.getProperty("hours"));
+    expectedTermRatingEntity.setProperty("difficulty", 4);
```

Comment:
You will likely get an error thrown here and on all number comparisons because datastore stores them as Longs and when you just put 4 I think its made an int. So you need to case the 4 as a Long